### PR TITLE
add curvefs proto

### PR DESCRIPTION
**Pull request by cw123**
### What problem does this PR solve?

Problem Summary: add fs proto

### What is changed and how it works?

What's Changed: add fs proto

How it Works: add fs proto

Side effects(Breaking backward compatibility? Performance regression?):  NO

### Check List

- [x] Relevant documentation/comments is changed or added
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Review by xu-chaojie — APPROVED** [review]

File: curvefs/proto/metaserver.proto [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    required MetaStatusCode statusCode = 1;
+}
+
+// inode interface
+message GetInodeRequest {
+    required uint32 fsId = 1;
+    required uint64 inodeId = 2;
+}
+
+enum FileType {
```
Comment (xu-chaojie) [inline curvefs/proto/metaserver.proto:90]: conflict with 'FileType' in 'include/client/libcurve.h'

Comment (cw123) [reply curvefs/proto/metaserver.proto]: chenge FileType to FsFileType

File: curvefs/proto/metaserver.proto [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    required uint32 ctime = 4;
+    required uint32 mtime = 5;
+    required uint32 atime = 6;
+    required uint32 uid = 7;
+    required uint32 gid = 8;
+    required uint32 mode = 9;
+    required sint32 nlink = 10;
+    required FsFileType type = 11;
+    optional string symlink = 12;   // TYPE_SYM_LINK only
+    repeated VolumeExtent volumeExtents = 13;  // TYPE_FILE only
```
Comment (xu-chaojie) [inline curvefs/proto/metaserver.proto:116]: use 'VolumeExtentList' will be better，avoid copy

**Review by xu-chaojie — CHANGES_REQUESTED** [review]

**Review by baijiaruo — APPROVED** [review]

**Review by wu-hanqing — APPROVED** [review]

File: curvefs/proto/metaserver.proto (hunk 2 above)
Comment (cw123) [reply curvefs/proto/metaserver.proto:116]: done

File: curvefs/proto/metaserver.proto [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+message GetDentryResponse {
+    required MetaStatusCode statusCode = 1;
+    optional Dentry dentry = 2;
+}
+
+message ListDentryRequest {
+    required uint32 fsId = 1;
+    required uint64 dirInodeId = 2;
+    optional uint32 last = 3;     // require dentry from last count
```
Comment (xu-chaojie) [inline curvefs/proto/metaserver.proto:49]: the 'last' dentry must be a name

**Review by xu-chaojie — CHANGES_REQUESTED** [review]

File: curvefs/proto/metaserver.proto (hunk 3 above)
Comment (cw123) [reply curvefs/proto/metaserver.proto]: done

**Review by xu-chaojie — COMMENTED** [review]
ok

**Review by xu-chaojie — APPROVED** [review]

**Comment by cw123** [comment]
recheck

**Pull request merged**